core/src/tangram.cpp

```diff
@@ -109,7 +109,10 @@ Map::~Map() {
     // The unique_ptr to Impl will be automatically destroyed when Map is destroyed.
     impl->tileWorker.stop();
     impl->asyncWorker.reset();
-    impl->jobQueue.runJobs();
+
+    // Make sure other threads are stopped before calling stop()!
+    // All jobs will be executed immediately on add() afterwards.
+    impl->jobQueue.stop();
```

Comment:
Future bugs :) This should make things more robust though. E.g. in our current use Impl::renderState could be destructed before Impl::jobQueue without bugs.

The actual bug I was looking at was the static UrlWorkers still holding onto RasterTileTask-\>RasterSource-\>Textures where the texture disposer called into garbage.